go-cosmwasm/types/stderror.go

```diff
@@ -62,7 +62,7 @@ type GenericErr struct {
 }
 
 func (e GenericErr) Error() string {
-	return fmt.Sprintf("generic: %s", e.Msg)
+	return fmt.Sprintf("encrypted: %s", e.Msg)
```

Comment:
I don't think this is right. The inner error will also be "encrypted" if you `.Error()` it.

Reply:
Isn't this already the inner error? That's why I changed it in the first place

Reply:
In order to keep the go side happy, when we encrypt the output we wrap any error within a generic error.
This:
```json
{ "generic_err": { "msg": "bla bla" } }
```
Becomes this:
```json
{ "generic_err": { "msg": "{ \"generic_err\": { \"msg\": \"bla bla\" } }" } }
```
Becomes this:
```json
{ "generic_err": { "msg": "1GGPyFBJVaHRVnoc+0WP2Maf9ulTqCStJsVsDuj64PsyjaUEbldM94+z/dW730nOhFFaSnmmoEw=" } }
```
After decryption on the go side, we get the original error which is `generic_err`, but when we `.Error()` it we'll get this `encrypted:` string instead of the `generic:` string.

It's not that big of a deal...

Reply:
(Also lets pretend that the random bytes here weren't generated by `cat /dev/urandom | head -c 56 | base64`)

Reply:
Hmm... I changed it specfically because when outputting to the chain to change:

``` "raw_log": "execute wasm contract failed: generic: Jz0Nt/f/ZH9VWi1KQOL/WlWJ2ThcqFGFiEZ/1DOU2tEx40ubeWpGAHXtp3xb/ipBTGi/rQ==: failed to execute message; message index: 0",```

to:

``` "raw_log": "execute wasm contract failed: encrypted: Jz0Nt/f/ZH9VWi1KQOL/WlWJ2ThcqFGFiEZ/1DOU2tEx40ubeWpGAHXtp3xb/ipBTGi/rQ==: failed to execute message; message index: 0",```

Is there any other case where it would make things less clear? I tried searching the logs in the integration tests but didn't see another example

Reply:
Probably CosmWasm are using this to show errors to users on the CLI.
Can you search where `StdError` is returned from the keeper and what the handler/CLI is doing with it?
It's most probably just returned as a string to the user, so I'd say I'm fine with it. :shrug:

Reply:
Didn't find anything except returning as a string. Worst comes to worst we can change this later on as well